Firestore/Example/Firestore_Benchmarks_iOS/FSTLevelDBBenchmarkTests.mm

```diff
@@ -0,0 +1,180 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import "../../../../../../../Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks/XCTest.framework/Headers/XCTest.h"
+
+#include "benchmark/benchmark.h"
+
+#import "Firestore/Source/Core/FSTTypes.h"
+#import "Firestore/Source/Local/FSTLevelDB.h"
+#import "Firestore/Source/Local/FSTLevelDBKey.h"
+#import "Firestore/Source/Local/FSTLocalSerializer.h"
+#import "Firestore/Source/Model/FSTDocumentKey.h"
+#import "Firestore/Source/Remote/FSTSerializerBeta.h"
+#include "Firestore/core/src/firebase/firestore/local/leveldb_transaction.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+using firebase::firestore::local::LevelDbTransaction;
+using firebase::firestore::model::DatabaseId;
+
+// Pre-existing document size
+static const int kDocumentSize = 1024 * 2;  // 2 kb
+
+static std::string DocumentData() {
+  return std::string(kDocumentSize, 'a');
+}
+
+static std::string UpdatedDocumentData(int documentSize) {
+  return std::string(documentSize, 'b');
+}
+
+static NSString *LevelDBDir() {
+  NSError *error;
+  NSFileManager *files = [NSFileManager defaultManager];
+  NSString *dir =
+      [NSTemporaryDirectory() stringByAppendingPathComponent:@"FSTPersistenceTestHelpers"];
+  if ([files fileExistsAtPath:dir]) {
+    // Delete the directory first to ensure isolation between runs.
+    BOOL success = [files removeItemAtPath:dir error:&error];
+    if (!success) {
+      [NSException raise:NSInternalInconsistencyException
+                  format:@"Failed to clean up leveldb path %@: %@", dir, error];
+    }
+  }
+  return dir;
+}
+
+static FSTLevelDB *LevelDBPersistence() {
+  // This owns the DatabaseIds since we do not have FirestoreClient instance to own them.
+  static DatabaseId database_id{"p", "d"};
+
+  NSString *dir = LevelDBDir();
+  FSTSerializerBeta *remoteSerializer = [[FSTSerializerBeta alloc] initWithDatabaseID:&database_id];
+  FSTLocalSerializer *serializer =
+      [[FSTLocalSerializer alloc] initWithRemoteSerializer:remoteSerializer];
+  FSTLevelDB *db = [[FSTLevelDB alloc] initWithDirectory:dir serializer:serializer];
+  NSError *error;
+  BOOL success = [db start:&error];
+  if (!success) {
+    [NSException raise:NSInternalInconsistencyException
+                format:@"Failed to create leveldb path %@: %@", dir, error];
+  }
+
+  return db;
+}
+
+class LevelDBFixture : public benchmark::Fixture {
+  virtual void SetUp(benchmark::State &state) {
```

Comment:
Can you replace `virtual` with `override`? `void SetUp(benchmark::State &state) override` (similarly with `TearDown`).

Reply:
Done.